numerapi/numerapi.py

```diff
@@ -1103,3 +1103,19 @@ def daily_submissions_performances(self, username: str) -> List[Dict]:
         performances = [p for p in performances
                         if any([p['correlation'], p['fnc'], p['mmc']])]
         return performances
+
+    def get_dataset_list(self):
+        query = """
+            query { listDatasets }
+        """
+        return self.raw_query(query)['data']['listDatasets']
+
+    def download_dataset(self, filename: str, dest_path: str):
+        query = """
+        query ($filename: String!) {
+            dataset(filename: $filename)
+        }
+        """
+        dataset_url = self.raw_query(query, {'filename': filename}, authorization=True)['data']['dataset']
```

Comment:
Does it really need authorization to get the dataset url?

Reply:
yep - it does for now

Reply:
ok, makes sense